pubsub/gcppubsub/gcppubsub.go

```diff
@@ -210,8 +211,27 @@ func (o *URLOpener) OpenTopicURL(ctx context.Context, u *url.URL) (*pubsub.Topic
 
 // OpenSubscriptionURL opens a pubsub.Subscription based on u.
 func (o *URLOpener) OpenSubscriptionURL(ctx context.Context, u *url.URL) (*pubsub.Subscription, error) {
-	for param := range u.Query() {
-		return nil, fmt.Errorf("open subscription %v: invalid query parameter %q", u, param)
+	// Set subscription options to use defaults
+	o.SubscriptionOptions = SubscriptionOptions{
```

Comment:
This doesn't look right, because 
a) it will override any `SubscriptionOptions` that the user actually provided if they are using a custom `Opener`.
b) Down below, the struct is updated based on the URL parameter; since we pass a pointer to the struct to `OpenSubscription`, that essentially updates all of the subscribers to whatever the latest URL parameter was.

I think this should clone the struct and update the `MaxBatchSize` based on the URL parameter. There's no need to worry about the default here since it is handled later.

Something like:

```
opts := o.SubscriberOptions // clone the options
```

and then update `ops.MaxBatchSize` down below as well.